src/concurrency/sync.rs

```diff
@@ -227,7 +241,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
     /// - If yes, fetches the data from `alloc_extra.sync`, or calls `missing_data` if that fails
     ///   and stores that in `alloc_extra.sync`.
     /// - Otherwise, calls `new_data` to initialize the primitive.
```

Comment:
```suggestion
    /// - Otherwise, calls `new_data` to initialize the primitive.
    ///
    /// The return value is a *clone* of the stored data, so if you intend to mutate it
    /// better wrap everything into an `Rc`.
```